hypershift-operator/controllers/nodepool/aws.go

```diff
@@ -147,10 +142,19 @@ func awsMachineTemplateSpec(infraName string, hostedCluster *hyperv1.HostedClust
 				RootVolume:               rootVolume,
 				AdditionalTags:           tags,
 				InstanceMetadataOptions:  instanceMetadataOptions,
-				Tenancy:                  tenancy,
 			},
 		},
 	}
+
+	if nodePool.Spec.Platform.AWS.Placement != nil {
+		awsMachineTemplateSpec.Template.Spec.Tenancy = nodePool.Spec.Platform.AWS.Placement.Tenancy
+
+		if capacityReservation := nodePool.Spec.Platform.AWS.Placement.CapacityReservation; capacityReservation != nil {
+			awsMachineTemplateSpec.Template.Spec.CapacityReservationID = ptr.To(capacityReservation.ID)
+			awsMachineTemplateSpec.Template.Spec.MarketType = capiaws.MarketTypeOnDemand
```

Comment:
the main use case this wants to support is ML gpu reservations. For that, marketType needs to be "capacity-block".
So we can either hardcode it and document that behaviour in the CapacityReservation.
Expose it as an enum and only allow "capacity-block". Same but also allow "on-demand"